pallets/process-validation/src/tests/create_process.rs

```diff
@@ -1,10 +1,142 @@
 use super::*;
+use crate::{Process, Version, ProcessStatus, Restriction::None, ProcessModel, VersionModel};
+use crate::Event::*;
+use frame_support::{assert_ok};
+use sp_std::prelude::*;
 
-use frame_support::assert_ok;
+// -- fixtures --
+#[allow(dead_code)]
+const PROCESS_ID1: [u8; 32] = [
+    1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+];
+const PROCESS_ID2: [u8; 32] = [
+    1, 2, 3, 5, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+];
 
 #[test]
-fn create_process_simple() {
+fn returns_error_if_origin_validation_fails_and_no_data_added() {
     new_test_ext().execute_with(|| {
-        assert_ok!(ProcessValidation::create_process(Origin::root()));
+        // update to check for error type
+        assert!(ProcessValidation::create_process(
+            Origin::none(),
+            PROCESS_ID1,
+            vec![{ None }]
+        ).is_err());
```

Comment:
I would recomment for this:

```rust
assert_noop!(
  ProcessValidation::create_process(Origin::none(), PROCESS_ID1, vec![{ None }],
  DispatchError::BadOrigin
);
```

As used in the `symmetric_key` pallet. That'll also assert that nothing has changed in storage I believe

Reply:
Nice, event better, just added some errors, updating

Reply:
refactored, pushing shortly